src/Service/Models/MagicLinkTokenRequest.cs

```diff
@@ -0,0 +1,24 @@
+using System.Net.Mail;
+
+namespace Passwordless.Service.Models;
+
+/// <summary>
+/// Used to generate and send out the magic link email.
+/// </summary>
+/// <param name="userId">User identifier for the recipient of the magic link.</param>
+/// <param name="emailAddress">Email address for the intended user.</param>
+/// <param name="linkTemplate">Template used for creating the magic link. Token template string will be swapped with actual token.</param>
+/// <param name="timeToLive">Time span the magic link will be valid for. Default lifespan is 15 minutes.</param>
+public class MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate, TimeSpan timeToLive) : RequestBase
+{
+    public MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate)
+        : this(userId, emailAddress, linkTemplate, new TimeSpan(0, 15, 0)) { }
+    public MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate, int timeToLive)
+        : this(userId, emailAddress, linkTemplate, new TimeSpan(0, 0, timeToLive)) { }
```

Comment:
Don't use `new TimeSpan(...)` like that. If `timeToLive` is above 60, then the above code will throw. Use `TimeSpan.FromSeconds(...)` etc.

Also, I'd argue we don't need the extra constructor for `int timeToLive` since we do the conversion in `SendMagicLinkRequest.ToDto()` already, and we can just do `TimeSpan.FromSeconds(...)` there, making things simpler.